src/main/java/bio/terra/cda/app/generators/QuerySqlGenerator.java

```diff
@@ -59,32 +59,23 @@ protected String generate() throws IllegalArgumentException {
 
     TableInfo tableInfo = dataSetInfo.getTableInfoFromField(body);
     String tableName = tableInfo.getTableName();
-    List<String> whereClauses = new ArrayList<>();
-    JoinBuilder jb = new JoinBuilder();
-    List<Join> pathToSystem = Collections.emptyList();
+    String identifierTablePrefix = tableName; // just a default - will validate below
+    String whereClause = "";
+    String fk = "";
 
     if (system != null && system.length() > 0) {
       String systemParam = this.parameterBuilder.addParameterValue("text",system);
       String toTable = tableName + "_identifier";
       if (dataSetInfo.getTableInfo(toTable) == null) {
         toTable = "subject_identifier";
+        identifierTablePrefix = "subject";
+        fk = "cda_subject_id";
+      } else {
+        fk = tableInfo.getPrimaryKeys().get(0).getName();
       }
-      pathToSystem = jb.getPath(tableName, toTable, "system", SqlUtil.JoinType.LEFT);
-
-
-      QueryField systemField =
-          queryFieldBuilder.fromPath( toTable + "_system");
-      whereClauses.add(systemField.getName() + " = " + systemParam);
-    }
-
-    String whereStr = "";
-    if (!whereClauses.isEmpty()) {
-      whereStr = " WHERE " + String.join(" AND ", whereClauses);
+      whereClause = String.format(" WHERE %s IN (SELECT DISTINCT(%s_id) FROM %s WHERE system = %s)", fk, identifierTablePrefix, toTable, systemParam);
```

Comment:
Will this work for somatic_mutation?

Reply:
yes, I added some comments to indicate how we handle that case